build_scripts/build_collect_diff.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+#
+# Script to build the kernel, collect compiled file lists using modified kernel scripts,
+# and generate a git diff report based on the collected lists.
+
+set -e
+
+# check and install gcc-11 if not already installed
+install_package_safe() {
+    if ! command -v libssl-dev &> /dev/null; then
+        sudo apt-get update
+        sudo apt-get install -y libssl-dev
+    else
+        echo "libssl-dev is already installed."
+    fi
+}
+
+# safely apply a patch to linux kernel
+apply_patch() {
+    # shellcheck disable=SC2154
+    local patch_path="$root_path/scripts/change-impact-tools/fixdep-patch.file"
+
+    # Stash any changes if there is any
+    if ! git diff --quiet; then
+        git stash
+    fi
+
+    # Abort `git am` only if there is a patch being applied
+    if git am --show-current-patch &> /dev/null; then
+        git am --abort
+    fi
+    echo "path check: $(pwd)"
+    git apply < "$patch_path"
+    echo "applied the git patch"
+    echo "path check: $(pwd)"
+}
+
+# parse the JSON file
```

Comment:
Comments should generally be very rare and, in particular, not repeat function names. Easy to read code with no comments is far better than detailed comments that will quickly go out of date as the code evolves.

Reply:
Yes. Have already removed this comment.